console/readdvd/dvd_udf.cpp

```diff
@@ -729,178 +738,172 @@ static int UDFScanDir( udf_t *udf, struct AD Dir, char *FileName,
 
   /* Scan dir for ICB of file */
   lbnum = partition->Start + Dir.Location;
-    
-//  if(DVDUDFCacheLevel(device, -1) > 0) {
-    /* caching */
-      
-    //if(!GetUDFCache(device, LBUDFCache, lbnum, &cached_dir)) {
-    if(!GetUDFCache(udf, LBUDFCache, lbnum, &cached_dir)) {
-      dir_lba = (Dir.Length + DVD_VIDEO_LB_LEN) / DVD_VIDEO_LB_LEN;
-      //if((cached_dir = dvdalign_lbmalloc(device, dir_lba)) == NULL) {
-      if((cached_dir = (uint8_t*) malloc(2048 * dir_lba)) == NULL) {
-        return 0;
-      }
-      //if( DVDReadLBUDF( device, lbnum, dir_lba, cached_dir, 0) <= 0 ) {
-      if( read( udf->dev, cached_dir, lbnum, dir_lba, 0) ) {
-        //dvdalign_lbfree(device, cached_dir);
-		free(cached_dir);
-        cached_dir = NULL;
-      }
-      SetUDFCache(udf, LBUDFCache, lbnum, &cached_dir);
-    } else {
-      in_cache = 1;
-    }
-      
-    if(cached_dir == NULL) {
+
+  //  if(DVDUDFCacheLevel(device, -1) > 0) {
+  /* caching */
+
+  // if(!GetUDFCache(device, LBUDFCache, lbnum, &cached_dir)) {
+  if (!GetUDFCache(udf, LBUDFCache, lbnum, &cached_dir)) {
+    dir_lba = (Dir.Length + DVD_VIDEO_LB_LEN) / DVD_VIDEO_LB_LEN;
+    // if((cached_dir = dvdalign_lbmalloc(device, dir_lba)) == NULL) {
+    if ((cached_dir = (uint8_t *)malloc(2048 * dir_lba)) == NULL) {
       return 0;
     }
-      
-    p = 0;
-      
-    while( p < Dir.Length ) {
-      UDFDescriptor( &cached_dir[ p ], &TagID );
-      if( TagID == 257 ) {
-        p += UDFFileIdentifier( &cached_dir[ p ], &filechar,
-                                filename, &tmpICB );
-        if(cache_file_info && !in_cache) {
-          uint8_t tmpFiletype;
-          struct AD tmpFile;
-            
-          if( !strcasecmp( FileName, filename ) ) {
-            *FileICB = tmpICB;
-            found = 1;
-              
-          }
-          UDFMapICB(udf, tmpICB, &tmpFiletype,
-                    partition, &tmpFile);
-        } else {
-          if( !strcasecmp( FileName, filename ) ) {
-            *FileICB = tmpICB;
-            return 1;
-          }
+    // if( DVDReadLBUDF( device, lbnum, dir_lba, cached_dir, 0) <= 0 ) {
+    if (read(udf->dev, cached_dir, lbnum, dir_lba, 0)) {
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6072101)